doc/spec/namespace.md

```diff
@@ -0,0 +1,198 @@
+# Namespace
+The global namespace represents the full set of named repositories which can be
+referenced by Docker. Any scope within the namespace may refer to a subset of
+the named repos. A repository contains a set of content-addressable blobs and
+tags referencing those blobs. The namespace can be used to discover the
+location and certificate information of a repository based on DNS, HTTP
+requests and other means.
+
+A repository should always be referenced by its fully qualified name. If a
+client presents a shortened name to a user, that name should be fully expanded
+based on rules defined by the client before contacting a remote server. When
+mirroring repositories, the original name for a repository must be used.
+Changing a repository name is equivalent to copying or moving and will not be
+referenced by the original repository.
+
+## Terminology
+
+ - *Global Namespace* - The full set of referenceable names.
+ - *Name* - A fully qualified string containing both the domain and resource name
+ - *Repository* - A collection of objects under the same name within the
+namespace.
+ - *Namespace* - also *Namespace Scope* - A collection of repositories with a
+common name prefix and set of services including registry api, index, and trust
+context.
+ - *Short Name* - A name which does not contain a domain and requires
+expansion to a fully qualified name before resolving to a repository.
+
+## Format
+A name consists of two parts, a DNS host name plus a repository path. The host
```

Comment:
"repository path" is not the best description. We want to consider the "resource name" to be the full name and using the term "path" to refer to everything after the DNS name. This is mostly used to describe the format of the repository name and how it will relate to a discovery which requires DNS resolution.